crypto/src/merkle/mod.rs

```diff
@@ -6,8 +6,17 @@
 use crate::{errors::MerkleTreeError, hash::Hasher};
 use core::slice;
 use math::log2;
+
+#[cfg(feature = "alloc")]
```

Comment:
If we have `utils::collections` module as described in the other comment, we could replace this with `use utils::collections::*;`